candle-examples/examples/debertav2/main.rs

```diff
@@ -0,0 +1,397 @@
+            let input_ids = model_input.input_ids.to_vec2::<u32>()?;
+            let mut results: Vec<Vec<NERItem>> = Default::default();
+
+            for (input_row_idx, input_id_row) in input_ids.iter().enumerate() {
+                let mut current_row_result: Vec<NERItem> = Default::default();
+                let current_row_encoding = model_input.encoding.get(input_row_idx).unwrap();
+                let current_row_tokens = current_row_encoding.get_tokens();
+                let current_row_max_scores = max_scores_vec.get(input_row_idx).unwrap();
+
+                for (input_id_idx, _input_id) in input_id_row.iter().enumerate() {
+                    // Do not include special characters in output
+                    if current_row_encoding.get_special_tokens_mask()[input_id_idx] == 1 {
+                        continue;
+                    }
+
+                    let max_label_idx = max_indices_vec
+                        .get(input_row_idx)
+                        .unwrap()
+                        .get(input_id_idx)
+                        .unwrap();
+
+                    let label = id2label.get(max_label_idx).unwrap().clone();
+
+                    // Do not include those labeled as "O" ("Other")
+                    if label == "O" {
+                        continue;
+                    }
+
+                    current_row_result.push(NERItem {
+                        entity: label,
+                        word: current_row_tokens[input_id_idx].clone(),
+                        score: current_row_max_scores[input_id_idx].clone(),
+                        start: current_row_encoding.get_offsets()[input_id_idx].0,
+                        end: current_row_encoding.get_offsets()[input_id_idx].1,
+                        index: input_id_idx,
+                    });
+                }
+
+                results.push(current_row_result);
+            }
+
+            println!("\n{:?}", results);
+        }
+
+        TaskType::TextClassification(classification_model) => {
+            let inference_time = std::time::Instant::now();
+            let logits = classification_model.forward(
+                &model_input.input_ids,
+                Some(model_input.token_type_ids),
+                Some(model_input.attention_mask),
+            )?;
+
+            println!("Inferenced inputs in {:?}", inference_time.elapsed());
+
+            let predictions = logits.argmax(1)?.to_vec1::<u32>()?;
+            let scores = softmax(&logits, 1)?.max(1)?.to_vec1::<f32>()?;
+            let mut results = Vec::<TextClassificationItem>::default();
+
+            for (idx, prediction) in predictions.iter().enumerate() {
+                results.push(TextClassificationItem {
+                    label: id2label[prediction].clone(),
+                    score: scores[idx],
+                });
+            }
+
+            println!("\n{:?}", results);
+        }
+    }
+    Ok(())
+}
+
+fn create_benchmark<F>(
```

Comment:
I saw that `criterion` is in the workspace Cargo.toml file as a dependency, but I did not see it in the `candle-examples` crate. I wasn't sure if the inclusion of criterion in the examples was warranted or wanted.

Reply:
Current version is fine, we prefer not having too much dependencies on the examples so avoid using criterion there.

Reply:
That's what I was assuming. I don't even know what really compelled me to have a benchmark in there, but I figured it'd be good to have to compare against any Python version.